src/error.c

```diff
@@ -0,0 +1,46 @@
+#include "error.h"
+
+#include "tlocal.h"
+
+static TLOCAL MTY_Queue *Q = NULL;
+
+void error_local_init(void)
+{
+	error_local_clear();
+
+	Q = MTY_QueueCreate(10, sizeof(MTY_Error));
```

Comment:
so, here we are saying that each thread has a 10 error code history, shared among all "modules". once there are 10 errors and we haven't gone in to drain the history, errors 11+ will be lost.

what's more traditional is for each "module" to have one error code per "context," such that whether there is 1 context or 20 contexts, they don't compete to store error codes and potentially lose to each other. as for the separate issue of there being a FIFO of error codes per module, that isn't a thing i've seen before.